### PR TITLE
General Refactor

### PR DESCRIPTION
**Pull request by stefanandy**
General refactor and renaming of various components and functions

### CODE REVIEW

File: blockchain/interface.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package blockchain
 
 import "github.com/ElrondNetwork/elrond-sdk-erdgo/data"
 
-type addressHandler interface {
+type AddressHandler interface {
```
Comment (mateigraura) [inline blockchain/interface.go:5]: 💯

**Review by mateigraura — APPROVED** [review]
I think this pr should target `refactor-examples` branch, that also includes the examples refactor

File: blockchain/interface.go (hunk 1 above)
Comment (iulianpascalau) [reply blockchain/interface.go:5]: maybe add a comment?

File: tests/address_test.go [hunk 2]
```diff
@@ -45,9 +46,11 @@ func TestGenerateSameDNSAddress(t *testing.T) {
...
 	require.Nil(t, err)
 
-	newDNS, err := ag.CompatibleDNSAddressFromUsername("christopher.elrond")
+	newDNS, err := ag.CompatibleDNSAddressFromUsername("laura.elrond")
 	require.Nil(t, err)
 
 	fmt.Printf("Compatibile DNS address is %s\n", newDNS.AddressAsBech32String())
 	assert.Equal(t, "erd1qqqqqqqqqqqqqpgqp64e3pqxwwyy93t5wp2w2jnlf4lfx3ljqqgsh8qwvz", newDNS.AddressAsBech32String())
+
+	fmt.Println(hex.EncodeToString([]byte("borcan.elrond")))
```
Comment (iulianpascalau) [inline tests/address_test.go:55]: please remove this

**Pull request merged**